android/src-huawei/com/badlogic/gdx/pay/app/AndroidLauncher.java

```diff
@@ -0,0 +1,118 @@
+        this.game = game;
+        super.initFlavor(this.game);
+
+        this.game.purchaseManager = new HuaweiPurchaseManager(this, this);
+    }
+
+    @Override
+    public void onRegionNotSupported() {
+        Gdx.app.log("HUAWEI", "REGION NOT SUPPORTED");
+    }
+
+    @Override
+    public void onLoginRequired() {
+        Gdx.app.log("HUAWEI", "LOGIN REQUESTED");
+    }
+
+    @Override
+    public void onIAPError(IapApiException exception) {
+        Gdx.app.error("HUAWEI", "IAP EXC", exception);
+    }
+
+    @Override
+    public void onError(Exception exception) {
+        Gdx.app.error("HUAWEI", "IAP ERR", exception);
+    }
+
+    @Override
+    public void onPurchaseResult(PurchaseIntentResult result) {
+        try {
+            result.getStatus().startResolutionForResult(this, this.PURCHASE_STATUS_RESULT_CODE);
+        } catch (IntentSender.SendIntentException e) {
+        }
+    }
+
+    @Override
+    public void onConsumedResult(ConsumeOwnedPurchaseResult result) {
+
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        switch (requestCode) {
+            case PURCHASE_STATUS_RESULT_CODE:
+                if (resultCode == RESULT_OK) {
+                    if (data == null) {
+                        Log.e("onActivityResult", "data is null");
+                        return;
+                    }
+                    PurchaseResultInfo purchaseResultInfo = Iap.getIapClient(this).parsePurchaseResultInfoFromIntent(data);
+                    switch (purchaseResultInfo.getReturnCode()) {
+                        case OrderStatusCode.ORDER_STATE_CANCEL:
+                            // User cancel payment.
+                            Gdx.app.log("ORDER_STATE", "CANCEL");
+                            break;
+                        case OrderStatusCode.ORDER_STATE_FAILED:
+                        case OrderStatusCode.ORDER_PRODUCT_OWNED:
+                            // to check if there exists undelivered products.
+                            Gdx.app.log("ORDER_STATE", "FAILED/OWNED");
+                            break;
+                        case OrderStatusCode.ORDER_STATE_SUCCESS:
+                            // pay success.
+                            String inAppPurchaseData = purchaseResultInfo.getInAppPurchaseData();
+
+                            try {
+                                InAppPurchaseData inAppPurchaseDataItem = new InAppPurchaseData(inAppPurchaseData);
+                                Transaction transaction = new Transaction();
+                                transaction.setIdentifier(inAppPurchaseDataItem.getProductId());
+                                Gdx.app.log("ORDER_STATE", "SUCCESS productId: " + transaction.getIdentifier());
+
+                                //TODO: REMOVE THE FOLLOWING LINES, ARE ONLY TO FORCE THE CONSUMABLE TEST
```

Comment:
please remove the TODO, this way it looks like the code is still not finished. I am not sure if this is here as a hint for testing, or if there are still changes needed.